plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/IcebergRegisterTableUtil.java

```diff
@@ -0,0 +1,119 @@
+{
+    private IcebergRegisterTableUtil() {}
+
+    public static String getLatestMetadataLocation(TrinoFileSystem fileSystem, String location)
+    {
+        List<String> latestMetadataLocations = new ArrayList<>();
+        String metadataDirectoryLocation = format("%s/%s", stripTrailingSlash(location, "table_location"), METADATA_FOLDER_NAME);
+        try {
+            int latestMetadataVersion = -1;
+            FileIterator fileIterator = fileSystem.listFiles(metadataDirectoryLocation);
+            while (fileIterator.hasNext()) {
+                FileEntry fileEntry = fileIterator.next();
+                if (fileEntry.path().contains(METADATA_FILE_EXTENSION)) {
+                    OptionalInt version = parseVersion(fileEntry.path());
+                    if (version.isPresent()) {
+                        int versionNumber = version.getAsInt();
+                        if (versionNumber > latestMetadataVersion) {
+                            latestMetadataVersion = versionNumber;
+                            latestMetadataLocations.clear();
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                        else if (versionNumber == latestMetadataVersion) {
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                    }
+                }
+            }
+            if (latestMetadataLocations.isEmpty()) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, "No versioned metadata file exists at location: " + metadataDirectoryLocation);
+            }
+            if (latestMetadataLocations.size() > 1) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, format(
+                        "More than one latest metadata file found at location: %s, latest metadata files are %s",
+                        metadataDirectoryLocation,
+                        latestMetadataLocations));
+            }
+        }
+        catch (IOException e) {
+            throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed checking table's location: " + location, e);
+        }
+        return getOnlyElement(latestMetadataLocations);
+    }
+
+    public static String stripTrailingSlash(String value, String name)
+    {
+        checkNotEmpty(value, name);
+        return value.replaceFirst("/+$", "");
+    }
+
+    private static String checkNotEmpty(String value, String name)
+    {
+        if (value == null) {
+            throw new NullPointerException(name + " is null");
+        }
+        if (value.isEmpty()) {
+            throw new IllegalArgumentException(name + " is empty");
+        }
+        return value;
+    }
+
+    public static void validateMetadataFileName(String fileName)
+    {
+        if (fileName == null) {
+            // Metadata file name is optional, So don't validate for null
+            return;
+        }
+        String metadataFileName = fileName.trim();
```

Comment:
Why trim?

Reply:
To handle the case when user has provided the filename with prefix whitespace or trailing whitespace. Is this make sense to handle the case?

Reply:
Removed